include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp

```diff
@@ -252,19 +254,27 @@ class alignment_executor_two_way
         // Reset the get pointer.
         setg(std::ranges::begin(buffer), std::ranges::end(buffer));
 
+        using std::get;
+
         // Apply the alignment execution.
         size_t count = 0;
         size_t buffer_limit = in_avail();
-        for (; count < buffer_limit && !is_eof(); ++count, ++resource_it, ++gptr)
+        assert(buffer_limit >= chunk_size());
+
+        while (count < buffer_limit && !is_eof())
         {
-            auto && [tpl, idx] = *resource_it;
-            auto && [first_seq, second_seq] = tpl;
-            buffer_pointer write_to = gptr;
-            exec_handler.execute(kernel,
-                                 idx,
-                                 first_seq | views::all,
-                                 second_seq | views::all,
-                                 [write_to] (auto && res) { *write_to = std::move(res); });
+            for (auto && [sequence_pair, idx] : *resource_it)
+            {
+                buffer_pointer write_to = gptr;
```

Comment:
Why not 
```cpp
auto && [first_seq, second_seq] = sequence_pair;
```

Then you do not need the `std::get` and it gets more readable (at least for me).

Reply:
There is no special reason. The point is that the bind variables are not really needed more than once, so not unpacking them has at least a  message: "Don't spoil the scope with something you don't need." 
But once all of the PRs dealing with the adaption of ranges have been resolved the entire inner loop will vanish. So I guess it is not urgent to change that. Would that be ok for you?

Reply:
ok